src/devices/bus/isa/lbaenhancer.cpp

```diff
@@ -0,0 +1,150 @@
+|                                  J3,J2,J1 |
+|                      PAL16V8              |
+|                                           |
+|                                           |
+|---|                                   |---|
+    |-----------------------------------|
+
+Notes:
+
+J3, J2, J1 set BIOS (start) address to C800, CC00, D000, D400, D800 or DC00
+
+- one CD74LS245 
+- one PALCE16V8H-25
+- one 32 K EPROM : 27C256 (STM) (mirrored after 16 K)
+
+*/
+
+#include "emu.h"
+#include "lbaenhancer.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(ISA8_LBA_ENHANCER, lba_enhancer_device, "lba_enhancer", "VS Systems LBA Enhancer BIOS 1995")
+
+//-------------------------------------------------
+//  ROM( lba_enhancer )
+//-------------------------------------------------
+
+ROM_START( lbabios )
+	ROM_REGION( 0xc8000, "lbabios", 0 )
+	ROM_LOAD( "lbaenhancer.bin", 0x0000, 0x04000, CRC(39d4566d) SHA1(d275193a870250f212dc29768d4e68fb43770e95) )
+ROM_END
+
+//-------------------------------------------------
+//  rom_region - device-specific ROM region
+//-------------------------------------------------
+
+const tiny_rom_entry *lba_enhancer_device::device_rom_region() const
+{
+	return ROM_NAME( lbabios );
+}
+
+//-------------------------------------------------
+//  device_add_mconfig - add device configuration
+//-------------------------------------------------
+MACHINE_CONFIG_START( lba_enhancer_device::device_add_mconfig )
+//	MCFG_DEVICE_ADD("lba_enhancer", lba_enhancer, 0)
+MACHINE_CONFIG_END
+
+
+//-------------------------------------------------
+//  jumpers  
+//-------------------------------------------------
+static INPUT_PORTS_START( lba_enhancer_dsw )
+
+	PORT_START("ROM_ADDRESS")
+	PORT_CONFNAME( 0x07, 0x00, "16 K ROM bank")
+	PORT_CONFSETTING(    0, "C8000 - CBFFF" )
+	PORT_CONFSETTING(    1, "CC000 - CFFFF" ) 
+	PORT_CONFSETTING(    2, "D0000 - D3FFF" ) 
+	PORT_CONFSETTING(    3, "D4000 - D7FFF" ) 
+	PORT_CONFSETTING(    4, "D8000 - DBFFF" ) 
+	PORT_CONFSETTING(    5, "DC000 - DFFFF" ) 
+INPUT_PORTS_END
+
+//-------------------------------------------------
+//  isa8_areplay_device - constructor
+//-------------------------------------------------
+ioport_constructor lba_enhancer_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME( lba_enhancer_dsw );
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+// Byte 3 in extra BIOS specifies 8 blocks of 512 byte (4K)
+void lba_enhancer_device::device_start()
+{
+	set_isa_device();
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+void lba_enhancer_device::device_reset()
+{
+	if(m_current_rom_start == 0 )
```

Comment:
Please fix the spacing here - it's inconsistent on the opening and closing parentheses.